lib/msf/core/post/windows/version.rb

```diff
@@ -0,0 +1,150 @@
+# -*- coding: binary -*-
+
+module Msf::Post::Windows::Version
+include Msf::Post::Windows::Registry
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Compat' => {
+          'Meterpreter' => {
+            'Commands' => %w[
+              stdapi_railgun_api
+            ]
+          }
+        }
+      )
+    )
+  end
+
+  def get_version_info
+    result = get_version_info_impl
+    if result.nil?
+      print_error("Couldn't retrieve the target's build number!")
+      raise RuntimeError.new("Couldn't retrieve the target's build number!")
+    end
+
+    result
+  end
+
+  def get_version_info_fallback_impl
+        build_num_raw = cmd_exec('ver')
+        groups = build_num_raw.match(/.*Version\s+(\d+)\.(\d+)\.(\d+)(\.(\d+))?/)
+        if groups.nil?
+          return nil
+        end
+
+        major, minor, build, unused, revision = groups.captures
+        revision = 0 if revision.nil?
+        # Default to workstation, since it'll likely be an older OS - pre Server editions
+        return Msf::WindowsVersion.new(major.to_i, minor.to_i, build.to_i, 0, Msf::WindowsVersion::VER_NT_WORKSTATION)
+  end
+
+  def get_version_info_impl
+    if session.type == 'meterpreter'
+      result = session.railgun.ntdll.RtlGetVersion(input_os_version_info_ex)
+      os_version_info_ex = unpack_version_info(result['VersionInformation'])
```

Comment:
Is there a possibility that Railgun raise an exception? or maybe `result` could be `nil`? I believe this should be handled to avoid breaking.

Reply:
It's certainly conceivable that railgun fails. Looking through the code, this could perhaps be either a `Rex::TimeoutError` (for a dead session), or a `RequestError` (e.g. extension failed to load); maybe some other ones - I'd have to look further.

Given it won't be a graceful handling (i.e. we'd need to throw an exception anyway), do you think it's much clearer handling and re-throwing?

I'm also a bit hesitant just because, looking at other usages of railgun throughout the codebase, I don't see that pattern anywhere else; most other places I looked let the exception bubble up, and handle it at the module level anyway.